packages/vue-language-core/src/utils/transform.ts

```diff
@@ -39,7 +39,10 @@ export function walkInterpolationFragment(
 	};
 	ast.forEachChild(node => walkIdentifiers(ts, node, varCb, localVars));
 
-	ctxVars = ctxVars.sort((a, b) => a.offset - b.offset);
+	// Avoid duplicate ctx vars
+	// Fixes https://github.com/vuejs/language-tools/issues/3258
+	const localVarsKeys = Object.keys(localVars);
+	ctxVars = ctxVars.filter(v => !localVarsKeys.includes(v.text)).sort((a, b) => a.offset - b.offset);
```

Comment:
Wouldn't it make sense to avoid pushing to `ctxVars` in the first place above?

Reply:
Makes sense. Update later

Reply:
Done